### PR TITLE
Fix for contains predicate

### PR DESCRIPTION
**Pull request by rjuliano**
*Issue #, if available: #458 

*Description of changes:*
The issue is that the `contains` predicate assumes that the field we're querying is a list. However, when used on a String field, it results in invalid SQL syntax. So for example:

```
Blog.TAGS.contains("something");
```

yields

```sql
"something" IN tags
```  
which results in an error.

At first, I thought fixing it would just be a matter of determining whether the field we’re querying against is a List<> or a String. However, that information is not available from inside the `SQLPredicate` class as it exists today, so I need to explore some options.

## Update
Decided to go with the [instr](https://www.sqlite.org/lang_corefunc.html#instr) function to handle string pattern matching operation.

------------

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: aws-datastore/src/main/java/com/amplifyframework/datastore/storage/sqlite/adapter/SQLPredicate.java [hunk 1]
```diff
@@ -141,12 +141,25 @@ private StringBuilder parsePredicateOperation(QueryPredicateOperation<?> operati
                         .append("?");
             case CONTAINS:
                 ContainsQueryOperator containsOp = (ContainsQueryOperator) op;
-                addBinding(containsOp.value());
-                return builder.append("?")
+                //TODO: Need to somehow the the underlying model type to determine whether the
```
Comment (rjuliano) [inline aws-datastore/src/main/java/com/amplifyframework/datastore/storage/sqlite/adapter/SQLPredicate.java:144]:
```suggestion
                //TODO: Need to somehow get the underlying model type to determine whether the
```

File: aws-datastore/src/test/java/com/amplifyframework/datastore/storage/sqlite/SQLPredicateTest.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
+package com.amplifyframework.datastore.storage.sqlite;
```
Comment (jamesonwilliams) [inline aws-datastore/src/test/java/com/amplifyframework/datastore/storage/sqlite/SQLPredicateTest.java:1]: Please add the copyright header

File: aws-datastore/src/test/java/com/amplifyframework/datastore/storage/sqlite/SQLPredicateTest.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+     * @throws DataStoreException
+     */
+    @Test
+    public void testContainsForStringList() throws DataStoreException {
+        QueryPredicateOperation<String> predicate = Blog.TAGS.contains("something");
+        SQLPredicate sqlPredicate = new SQLPredicate(predicate);
+        assertStringResults(sqlPredicate, "tags");
+    }
+
+    private void assertStringResults(SQLPredicate sqlPredicate, String fieldName) {
```
Comment (jamesonwilliams) [inline aws-datastore/src/test/java/com/amplifyframework/datastore/storage/sqlite/SQLPredicateTest.java:41]: Can we get a more descriptive name for this helper method?

**Comment by rjuliano** [comment]
Fix will be included in the next release.

**Pull request merged**